tests/test_multi_alternate_primary_failures.py

```diff
@@ -96,15 +96,14 @@ def test_003_002_stop_primary():
     assert node2.get_state().assigned == "primary"
     node2.fail()
 
-    # wait for node3 to become the new write node
-    assert node2.wait_until_assigned_state(target_state="demoted")
-    assert node3.wait_until_state(target_state="wait_primary")
+    # node3 can't be promoted when it's the only one reporting its LSN
+    assert node2.wait_until_assigned_state(target_state="draining")
+    assert node3.wait_until_state(target_state="report_lsn")
```

Comment:
I think it might be nice if we could wait a bit here, to ensure that `node3` doesn't become `wait_primary`.

Reply:
Something like:
```suggestion
    assert node3.wait_until_state(target_state="report_lsn")
    node3.sleep(10)
    assert node3.wait_until_state(target_state="report_lsn")
```